docs/source/named_tensor.rst

```diff
@@ -0,0 +1,302 @@
+.. currentmodule:: torch
+
+.. _named_tensors-doc:
+
+Named Tensors
+=============
+
+Named Tensors aim to make tensors easier to use by allowing users to associate
+explicit names with tensor dimensions. In most cases, operations that take
+dimension parameters will accept dimension names, avoiding the need to track
+dimensions by index. In addition, named tensors use names to automatically
```

Comment:
nit: you probably want to say "position" here.  "dimensions by position" or "dimensions by index position" sounds good.